File: java_project/Strings/ExtensionIsolator.java
package String;

public class ExtensionIsolator {
    public static void main(String[] args) {
        String s1 = "abhishek.forever3@.gmail.com";
        int dotIndex = -1;
        int length = s1.length();
         
        for (int i = length - 1; i >= 0; i--) {
            if (s1.charAt(i) == '.'){
                dotIndex = i;
                break;
            }
        
        }

        if (dotIndex== -1 || dotIndex== length -1){
            System.out.println("Error: No valid file extension found.");
        }
        else{
            String extension = "";
            for (int i=dotIndex + 1; i<length; i++){
                extension =    extension + s1.charAt(i);
            }
            System.out.println("Full Filename : " + s1);
            System.out.println("Dot Located At: Index " + dotIndex);
            System.out.println("Extension Only: " + extension);
        }
    }
}

File: js_project/Strings/ExtensionIsolator.js
let s1 = "abhishek.forever3@gmail.com";
let dotIndex = -1;
let length = s1.length;

for (let i = length-1; i >=0; i--){

    if (s1.charAt(i)=='.'){
        dotIndex = i;
        break;
    }
}
if (dotIndex== -1 || dotIndex == length -1){
    console.log ("Error: No valid file extension found.");
}
else{
    let extension = "";

    for (let i=dotIndex+1 ; i<length; i++) {
        extension+=s1.charAt(i);
    }
    console.log("Full Filename : " + s1);
    console.log("Dot Located At: Index " + dotIndex);
    console.log("Extension Only: " + extension);
}